pkg/graph/builder.go

```diff
@@ -674,13 +687,102 @@ func extractDependencies(env *cel.Env, expression string, resourceNames []string
 	return dependencies, isStatic, nil
 }
 
+// setExpectedTypeOnDescriptor sets the ExpectedType field on a FieldDescriptor by:
+// 1. Taking the path (e.g., "spec.containers[0].name")
+// 2. Stripping array indices (e.g., "spec.containers.name")
+// 3. Constructing type name with __type_ prefix
+// 4. Looking up schema for that path
+// 5. Converting schema to CEL type
+func setExpectedTypeOnDescriptor(descriptor *variable.FieldDescriptor, rootSchema *spec.Schema, resourceID string) {
+	// String templates always expect StringType - already set by parser
+	if !descriptor.StandaloneExpression {
```

Comment:
feels weird to have 2 places where expectation types are set. is it worth unifying this to this place here?